### PR TITLE
Working Game Version - Needs further modification

### PR DESCRIPTION
**Pull request by imgeyuez**
A first working master version with correct logging playing etc.

Theree are some missing things:
- reprompting
- template checking needs to be commented in again (there is currently the issue that the messages from the players are cut. to check if the game runs otherwise, I commented the teemplate in the validation out)
- proper scoring metrics
- -clean up :D

### CODE REVIEW

File: games/datingsim/resources/prompts/further_prompts.template [hunk 1]
```diff
@@ -0,0 +1 @@
+$character_name 's Message: $response
```
Comment (JPasniczek) [inline games/datingsim/resources/prompts/further_prompts.template]: Please add end-line for this file <3

File: games/datingsim/master.py [hunk 2]
```diff
@@ -109,101 +159,163 @@ def setup(self, **game_instance) -> None:
...
+        response_pattern = r"\[response\](.+)"
+        response_match = re.search(response_pattern, answer, re.DOTALL)
+        last_message = response_match.group(1)
+        # print(f"Last message: {last_message}")
+        return last_message
+
 
+    
 
 # This needs to be adjusted or removed completely (replaced)
```
Comment (JPasniczek) [inline games/datingsim/master.py:320]: Can we delete this function? or should we refactor it?

File: games/datingsim/instancegenerator.py [hunk 3]
```diff
@@ -47,12 +48,14 @@ def on_generate(self):
         # predefine actions in case that we include them
         actions = None  # number of how often each player can say sth
         n_turns = 25
+        max_retries = 3
 
         # initial prompts for player A and player B
         # TO-DO: Change prompts
         initial_prompt_a = self.load_template('resources/initial_prompts/initialprompt_playerA.template') # self.load_template('C:/Users/imgey/Desktop/MASTER_POTSDAM/SoSe24/PM2/project/rizzSim/rizzSim/games/datingsim/resources/resources/initial_prompts/initial_pc_prompt.template')
```
Comment (JPasniczek) [inline games/datingsim/instancegenerator.py]: Do we need this commented out code?

**Review by JPasniczek — CHANGES_REQUESTED** [review]
Small changes requested but all in all amazing work <3

File: games/datingsim/master.py (hunk 2 above)
Comment (imgeyuez) [reply games/datingsim/master.py:320]: I would like to keep them commented in there for now until we have a working re-prompting and enforcing the template. As a guidence or reference while working on it. After that, we can gladly delete them.

**Review by JPasniczek — APPROVED** [review]
Thanks  for the minor fixes and amazing job! All Zucchini <3

**Pull request merged**